v2/pkg/subscraping/sources/dnsdb/dnsdb.go

```diff
@@ -58,42 +72,86 @@ func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Se
 		}
 
 		sourceName := s.Name()
+
+		path := fmt.Sprintf("lookup/rrset/name/*.%s", domain)
+		urlTemplate := fmt.Sprintf("%s/%s?", urlBase, path)
+		queryParams := url.Values{}
 		// ?limit=0 means DNSDB will return the maximum number of results allowed.
-		resp, err := session.Get(ctx, fmt.Sprintf(urlBase+"lookup/rrset/name/*.%s?limit=0&swclient=subfinder", domain), "", headers)
+		queryParams.Add("limit", "0")
+		queryParams.Add("swclient", "subfinder")
+
+		rate, err := getOffset(ctx, session, headers)
 		if err != nil {
 			results <- subscraping.Result{Source: sourceName, Type: subscraping.Error, Error: err}
 			s.errors++
-			session.DiscardHTTPResponse(resp)
 			return
 		}
 
-		scanner := bufio.NewScanner(resp.Body)
-		for scanner.Scan() {
-			line := scanner.Text()
-			if line == "" {
-				continue
-			}
-			var response safResponse
-			err = jsoniter.NewDecoder(bytes.NewBufferString(line)).Decode(&response)
+		var offsetMax uint64
+		if rate.OffsetMax.String() != rateNA {
+			offsetMax, err = strconv.ParseUint(rate.OffsetMax.String(), 10, 64)
 			if err != nil {
 				results <- subscraping.Result{Source: sourceName, Type: subscraping.Error, Error: err}
 				s.errors++
 				return
 			}
-			// Cond is a scalar enum of string values: {“begin”, “ongoing”, “succeeded”, “limited”, “failed”}
-			if (response.Cond == "" || response.Cond == "ongoing") && response.Obj.Name != "" {
-				results <- subscraping.Result{
-					Source: s.Name(), Type: subscraping.Subdomain, Value: strings.TrimSuffix(response.Obj.Name, "."),
+		}
+
+		var respCond string
+		for respCond == "" {
+			if rate.OffsetMax.String() != rateNA {
+				if s.results <= offsetMax {
+					queryParams.Set("offset", strconv.FormatUint(s.results, 10))
+				} else {
+					return
 				}
-				s.results++
-			} else if response.Cond == "failed" {
-				// DNSDB's terminating jsonl object's cond is "failed", notify the user.
-				err = fmt.Errorf("%s terminated with condition: %s",sourceName, response.Cond)
+			}
+			url := urlTemplate + queryParams.Encode()
+			//fmt.Fprintf(os.Stderr, "Making query: %s\n", url)
+
+			resp, err := session.Get(ctx, url, "", headers)
+			if err != nil {
 				results <- subscraping.Result{Source: sourceName, Type: subscraping.Error, Error: err}
 				s.errors++
+				session.DiscardHTTPResponse(resp)
+				return
+			}
+
+			scanner := bufio.NewScanner(resp.Body)
+			for scanner.Scan() {
+				line := scanner.Text()
+				if line == "" {
+					continue
+				}
+				var response safResponse
+				err = jsoniter.NewDecoder(bytes.NewBufferString(line)).Decode(&response)
+				if err != nil {
+					results <- subscraping.Result{Source: sourceName, Type: subscraping.Error, Error: err}
+					s.errors++
+					return
+				}
+				// Cond is a scalar enum of string values: {“begin”, “ongoing”, “succeeded”, “limited”, “failed”}
+				respCond = response.Cond
+				if (respCond == "" || respCond == "ongoing") && response.Obj.Name != "" {
```

Comment:
it seems like either respCond should be set to "" here to promote further outer loop execution, or the outerloop itself should be for respCond == "" || respCond == "ongoing" {